src/oData/jsonToOData.ts

```diff
@@ -215,18 +216,29 @@ export const convertToPineClientFilter = (
 	return handlePrimitiveFilter(parentKeys, filter);
 };
 
-export const orderbyBuilder = <T>(sortInfo: TableSortOptions<T> | null) => {
+export const orderbyBuilder = <T>(
+	sortInfo: TableSortOptions<T> | null,
+	customSort: AutoUIContext<T>['customSort'],
+) => {
 	if (!sortInfo) {
 		return null;
 	}
+
 	const { field, reverse, refScheme } = sortInfo;
 	if (!field) {
 		return null;
 	}
 	const direction = !reverse ? 'asc' : 'desc';
+	const customOrderByKey =
+		customSort?.[sortInfo.field as keyof typeof customSort];
+	if (typeof customOrderByKey === 'string') {
```

Comment:
Should we throw if this is not string | null?
Eg if the result is a function, as per `customSort?: Dictionary<(a: T, b: T) => number> | Dictionary<string>;`